contrib/NeonVM/stack.c

```diff
@@ -64,11 +66,27 @@ void pop(struct tagTStack *stack)
     cell_freeCell(p);
 }
 
-Cell *top(struct tagTStack *stack)
+Cell *top(TStack *stack)
 {
     if (isEmpty(stack)) {
         fatal_error("Stack underflow error.");
     }
 
     return stack->data[stack->top];
 }
+
+Cell *peek(TStack *stack, int element)
+{
+    if (isEmpty(stack)) {
+        fatal_error("Stack underflow error.");
+    }
+    if ((stack->top - element) < 0) {
+        fatal_error("Stack overflow error.");

```

Comment:
This is still an underflow situation, not an overflow.

Reply:
Fixed.